drivers/modem/modem_shell.c

```diff
@@ -0,0 +1,108 @@
+/** @file
+ * @brief Modem shell module
+ *
+ * Provide some modem shell commands that can be useful to applications.
+ */
+
+/*
+ * Copyright (c) 2018 Foundries.io
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <stdlib.h>
+#include <string.h>
+#include <device.h>
+#include <shell/shell.h>
+
+#include <misc/printk.h>
+
+#include <drivers/modem/modem_receiver.h>
+
+#define MODEM_SHELL_MODULE "modem"
+
+int modem_shell_cmd_list(int argc, char *argv[])
+{
+	struct mdm_receiver_context *mdm_ctx;
+	int i, count = 0;
+
+	printk("Modem receivers:\n");
+
+	for (i = 0; i < CONFIG_MODEM_RECEIVER_MAX_CONTEXTS; i++) {
+		mdm_ctx = mdm_receiver_context_from_id(i);
+		if (mdm_ctx) {
+			count++;
+			printk("%d:\tUART Name:    %s\n"
+				"\tManufacturer: %s\n"
+				"\tModel:        %s\n"
+				"\tRevision:     %s\n"
+				"\tIMEI:         %s\n"
+				"\tRSSI:         %d\n", i,
+			       mdm_ctx->uart_dev->config->name,
+			       mdm_ctx->data_manufacturer,
+			       mdm_ctx->data_model,
+			       mdm_ctx->data_revision,
+			       mdm_ctx->data_imei,
+			       mdm_ctx->data_rssi);
+		}
+	}
+
+	if (!count) {
+		printk("None found.\n");
+	}
+
+	return 0;
+}
+
+int modem_shell_cmd_send(int argc, char *argv[])
+{
+	struct mdm_receiver_context *mdm_ctx;
+	int ret, i, arg = 1;
+
+	/* list */
+	if (!argv[arg]) {
+		printk("Please enter a modem index\n");
+		return -EINVAL;
+	}
+
+	/* <index> of modem receiver */
+	i = atoi(argv[arg]);
```

Comment:
Better have "strtol(argv[arg], &endptr, 10);" instead of atoi() as the latter does not catch errors